app/assets/javascripts/controllers/tag_groups/controller.js

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import ColorPick from 'components/tag_groups/ColorPick';
+
+export default () => {
+  const bgColor = $('[name=bg-color]').data('current-color') || "#F17013";
+
+  ReactDOM.render(<ColorPick color ={ bgColor }/>, document.getElementById('tag_group_color'));
+
+  $(document).ready(function(){
+    $(document).bind('ajaxError', function(event, jqxhr, settings, exception){
+      $("#new_tag_group").render_form_errors( $.parseJSON(jqxhr.responseText) );
+    });
+  });
+
+  (function($) {
+
+    $.fn.modal_success = function(){
+      // close modal
+      this.modal('hide');
+
+      // clear form input elements
+      // todo/note: handle textarea, select, etc
+      this.find('form input[type="text"]').val('');
+
+      // clear error state
+      this.clear_previous_errors();
+    };
+
+    $.fn.render_form_errors = function(errors){
+      let $form = this,
+          model = $form.data('model');
+
+      $form.clear_previous_errors();
+
+
+      // show error messages in input form-group help-block
+      $.each(errors, function(field, messages){
+        let $input = $('input[name="' + model + '[' + field + ']"]');
+        $input
+          .closest('.form-group')
+          .addClass('has-error').find('.help-block').html( messages.join(' & ') );
+      });
+
+    };
+
+    $.fn.clear_previous_errors = function(){
+      $('.form-group.has-error', this).each(function(){
```

Comment:
We should avoid using CSS classes for selecting elements, CSS classes are meant to be used for styles. Add data attributes for that.